src/utils/hash-link-scroll.js

```diff
@@ -28,28 +35,29 @@ function hashLinkScroll() {
     if (hash !== '') {
       // Push onto callback queue so it runs after the DOM is updated
       setTimeout(() => {
+        let observer = null
+        let asyncTimerId = null
         const id = hash.replace('#', '')
         // since id will be encoded by browser,
         // decode it back as a string
-        const decodedID = decodeURIComponent(id)
+        const decodedId = decodeURIComponent(id)
         // Create a `MutationObserver` instance to watch for changes being made to the DOM tree.
         // Since `document.getElementById(id)` returns null when navigating from a different page 
         // without element of id, for example: When navigating from an article page to homepage
         // with hash `#categories` in url, `document.getElementById('categories')` returns null.
         // In such case, the `MutationObserver` instance keeps watching and will call it's callback 
         // when the DOM tree changes, page scrolls to the desired element as long as it can be gotten by id.
-        if (!getElementAndScroll(decodedID)) {
-          if (!observer) {
-            observer = new MutationObserver(() => { getElementAndScroll(decodedID) })
-          }
+        if (!getElementAndScroll(decodedId)) {
+          observer = new MutationObserver(() => { getElementAndScroll(decodedId, observer, asyncTimerId) })
```

Comment:
`getElementAndScroll ` -> `scrollToElement(element, observer, asyncTimeId)` ?

Reply:
Maybe we could remove `scrollToElement` function or rename it?
After second thoughts, `observer` and `asyncTimeId` is not necessary for `scrollToElement`.

Reply:
That's great!!
I move out `reset` function call so as not to send `observer` and `asyncTimeId` as params which are not necessary for `scrollToElement` indeed.

Please see commit `2d77509 `